strawberry/extensions/runner.py

```diff
@@ -21,37 +27,59 @@ def __init__(
 
     def _run_on_all_extensions(self, method_name: str, *args, **kwargs):
         for extension in self.extensions:
-            getattr(extension, method_name)(*args, **kwargs)
+            method = getattr(extension, method_name)
 
-    @contextmanager
-    def request(self):
-        self._run_on_all_extensions("on_request_start")
+            if inspect.iscoroutinefunction(method):
+                raise RuntimeError(SYNC_CONTEXT_WARNING)
```

Comment:
I added checks informing the user about async extension hooks being used during a synchronous schema execution. Not sure whether I like them. Any thoughts? 🤔